### PR TITLE
Delete Inventory Item through UI

### PR DESCRIPTION
**Pull request by zhengtr**
- Add delete UI in `rest_api.js`
- import and use `flask.make_response` in `routes.py`
- No `abort` is called so the message string should work fine

### CODE REVIEW

**Review by nishanthsanjeev — CHANGES_REQUESTED** [review]
Your changes are good - can you also add support for a 'Success' message to be displayed when we delete an Inventory item? In the given screenshot, note how there is a success message displayed when an item is created.

Screenshot 2023-04-14 at 5 59 04 PM

**Comment by sahildeshpande25** [comment]
@zhengtr can you merge master into your branch? My PR was just merged so you'll have to pull from master first.

**Comment by sahildeshpande25** [comment]
> Your changes are good - can you also add support for a 'Success' message to be displayed when we delete an Inventory item? In the given screenshot, note how there is a success message displayed when an item is created.
> 
There are flash messages supported on lines 100 and 104 but the delete function is outside the main function and can't access the flash message function as it is out of scope. You will have to move your function inside the main function.

**Comment by zhengtr** [comment]
> Your changes are good - can you also add support for a 'Success' message to be displayed when we delete an Inventory item? In the given screenshot, note how there is a success message displayed when an item is created.
> 
> Screenshot 2023-04-14 at 5 59 04 PM

I was trying to add it as well but couldn't figure out how. Would you please let me know how to do that?

File: service/static/js/rest_api.js [hunk 1]
```diff
@@ -77,3 +77,30 @@ $(function () {
...
+
+        ajax.done(function(res){
+            clear_form_data()
+            flash_message("Item has been Deleted!")
+        });
+
+        ajax.fail(function(res){
+            flash_message("Server error!")
+        });
+    });
```
Comment (sahildeshpande25) [inline service/static/js/rest_api.js:106]:
```suggestion

    // ****************************************
    // Delete a Pet
    // ****************************************

    $("#delete-btn").click(function () {

        let inventory_id = $("#inventory_id").val();

        $("#flash_message").empty();

        let ajax = $.ajax({
            type: "DELETE",
            url: `/inventory/${inventory_id}`,
            contentType: "application/json",
            data: '',
        })

        ajax.done(function(res){
            clear_form_data()
            flash_message("Item has been Deleted!")
        });

        ajax.fail(function(res){
            flash_message("Server error!")
        });
    });
}}
```

Comment (sahildeshpande25) [reply service/static/js/rest_api.js:106]: @zhengtr the suggested change should fix it

**Comment by zhengtr** [comment]
> > Your changes are good - can you also add support for a 'Success' message to be displayed when we delete an Inventory item? In the given screenshot, note how there is a success message displayed when an item is created.
> 
> There are flash messages supported on lines 100 and 104 but the delete function is outside the main function and can't access the flash message function as it is out of scope. You will have to move your function inside the main function.

Hey @sahildeshpande25  I am sorry that I didn't see this comment by you (otherwise I would know how to fix it without you providing the code details). 

I have fixed it and also merged with master. Please review @sahildeshpande25 @nishanthsanjeev

**Comment by sahildeshpande25** [comment]
@zhengtr while merging master into your branch, you have deleted the restock function in `rest_api.js`

**Comment by zhengtr** [comment]
> @zhengtr while merging master into your branch, you have deleted the restock function in `rest_api.js`

@sahildeshpande25 Some weird things happened when I applied suggested changes. It's fixed now.

**Review by nishanthsanjeev — APPROVED** [review]

**Review by sahildeshpande25 — APPROVED** [review]
Delete button works as expected.

**Pull request merged**